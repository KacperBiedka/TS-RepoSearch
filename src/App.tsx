import React, { FC } from 'react';
import './App.scss';
import { Route, BrowserRouter } from 'react-router-dom';
import { QueryParamProvider } from 'use-query-params';
import routes from './routes';



const App:FC = () => {
  const routeComponents = routes.map(({ path, component }, key) => {
    return (
      <QueryParamProvider ReactRouterRoute={Route} key={key}>
            <Route exact path={path} component={component} />
      </QueryParamProvider>
  )});
  return <BrowserRouter>{routeComponents}</BrowserRouter>;
}

export default App;
