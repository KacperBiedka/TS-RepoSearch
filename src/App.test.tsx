import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
test('renders the App component', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});