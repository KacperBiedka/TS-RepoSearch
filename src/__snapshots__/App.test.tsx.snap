// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the App component 1`] = `
<BrowserRouter>
  <QueryParamProvider
    ReactRouterRoute={[Function]}
    key="0"
  >
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
  </QueryParamProvider>
</BrowserRouter>
`;
